library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -34,6 +34,7 @@
     private static final String[] PROJECT_BATCH_ID = {DownloadContract.Batches._ID};
     private static final String WHERE_DELETED_VALUE_IS = DownloadContract.Batches.COLUMN_DELETED + " = ?";
     private static final String[] MARKED_FOR_DELETION = {"1"};
+    private static final String SORT_BY_ID_DESCENDING = DownloadContract.Batches._ID + " DESC";
```

Comment:
Ah I see the issue now. BUT we should not change this functionality because we need FIFO. 

Also surely if all but the first couldn't run in this new scenario, you'd still get 99 errors?